bundles/org.eclipse.passage.lic.api/src/org/eclipse/passage/lic/internal/api/version/SemanticVersion.java

```diff
@@ -0,0 +1,30 @@
+/*******************************************************************************
+ * Copyright (c) 2020 ArSysOp
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0/.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     ArSysOp - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.passage.lic.internal.api.version;
+
+/**
+ * Well structured representation of a version.
+ */
+public interface SemanticVersion {
+
+	int major();
+
+	int minor();
+
+	int service();
+
+	boolean hasQualifier();
+
+	String qualifier();
```

Comment:
What will be happen if one will call `qualifier()`  having `hasQualifier()==false`?

Reply:
This part is documented. Contract test will follow.